alica_engine/include/engine/model/TerminalState.h

```diff
@@ -12,19 +12,20 @@
 
 namespace alica {
 class PostCondition;
-
+class ModelFactory;
 /**
  * A terminal state within a plan. Indicates termination of the corresponding task
  */
 class TerminalState : public State {
```

Comment:
I guess final and overrides can be another PR entirely and for another time.

Reply:
Added override, but skip on final ;)